src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.domain.Fruit;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Fruit.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<Fruit.Operation, OperationHandler> operationHandlerMap) {
```

Comment:
It's a good practice to validate the input parameters of a constructor to avoid initializing your class with invalid arguments. Consider checking if the `operationHandlerMap` is not null or empty.